index_websocket.js

```diff
@@ -0,0 +1,97 @@
+const http = require("http");
+const open = require("open");
+const fs = require("fs");
+const WebSocketServer = require("websocket").server; // websocket server class
+
+let webSocket = null;
+// connection variable, to refer to the web socket connection to the client once connection is established
+let connection = null;
+// default port used if none is specified by the user
+let defaultPort = 9101;
+const host = "http://127.0.0.1";
+
+// check if a browser window is already open
+let isRunning = false;
+
+
+exports.visualise = function visualise(jsonData, port = defaultPort) {
+
+  // overhead of server creation only done on first call
+  if (!isRunning) {
+    isRunning = true;
+
+    // open index.html in the browser
+    open("http://127.0.0.1:" + port);
+
+    // create http Server
+    const httpServer = http.createServer((req, res) => {
+
+      // check if the which files to render
+      var url = req.url;
+
+      switch (url) {
+        case '/':
+          getStaticFileContent(res, 'public/index.html', 'text/html');
```

Comment:
Let's move the html file path to be a constant. The html file will exist in the same directory.